src/commands/transaction/reconstruct_transaction/mod.rs

```diff
@@ -0,0 +1,264 @@
+        };
+
+        eprintln!("\nArchive transaction:\n");
+        crate::common::print_unsigned_transaction(&prepopulated_transaction);
+        eprintln!();
+
+        let cmd = crate::commands::CliTopLevelCommand::Transaction(
+            crate::commands::transaction::CliTransactionCommands{
+                transaction_actions: Some(crate::commands::transaction::CliTransactionActions::ConstructTransaction(
+                    crate::commands::transaction::construct_transaction::CliConstructTransaction {
+                        sender_account_id: Some(signer_id.into()),
+                        receiver_account_id: Some(receiver_id.into()),
+                        next_actions: None
+                    }
+                ))
+            }
+        );
+        let mut cmd_cli_args = cmd.to_cli_args();
+
+        for archival_action in archival_actions {
+            let next_actions = crate::commands::transaction::construct_transaction::add_action_1::CliNextAction::AddAction(
+                crate::commands::transaction::construct_transaction::add_action_1::add_action::CliAddAction{
+                    action: action_transformation(archival_action)?
+                }
+            );
+            cmd_cli_args.extend(next_actions.to_cli_args());
+        }
+
+        let skip_action = crate::commands::transaction::construct_transaction::add_action_2::CliNextAction::Skip(
+            crate::commands::transaction::construct_transaction::skip_action::CliSkipAction{
+                network_config: Some(crate::commands::transaction::construct_transaction::skip_action::ClapNamedArgNetworkForTransactionArgsForSkipAction::NetworkConfig(
+                    crate::network_for_transaction::CliNetworkForTransactionArgs{
+                    network_name: Some(scope.network_name.clone()),
+                    transaction_signature_options: None
+                }
+                ))
+            }
+        );
+        cmd_cli_args.extend(skip_action.to_cli_args());
+
+        let near_cli_exec_path = crate::common::get_near_exec_path();
+        eprintln!("Here is your console command to run archive transaction. You can to edit it or re-run:");
+        eprintln!(
+            "{}\n",
+            shell_words::join(std::iter::once(near_cli_exec_path).chain(cmd_cli_args))
+        );
+
+        Ok(Self)
+    }
+}
+
+impl TransactionInfo {
+    fn input_network_name(
+        context: &crate::GlobalContext,
+    ) -> color_eyre::eyre::Result<Option<String>> {
+        crate::common::input_network_name(&context.config, &[])
+    }
+}
+
+fn action_transformation(
+    archival_action: Action,
+) -> color_eyre::eyre::Result<Option<crate::commands::transaction::construct_transaction::add_action_1::add_action::CliActionSubcommand>>{
+    match archival_action {
+        Action::CreateAccount(_) => {
+            Ok(Some(crate::commands::transaction::construct_transaction::add_action_1::add_action::CliActionSubcommand::CreateAccount(
```

Comment:
It will be easier to read if we `use crate::commands::transaction::construct_transaction::add_action_1::add_action::CliActionSubcommand` (don't use it globally,  add `use` at the beginning of this function)

Reply:
Done